In [1]:
import pandas as pd
import numpy as np


### Import dataset

In [6]:
df = pd.read_csv('/content/Wholesale customers data.csv')

In [7]:
df

Channel  Region  Fresh   Milk  Grocery  Frozen  Detergents_Paper  \
0          2       3  12669   9656     7561     214              2674   
1          2       3   7057   9810     9568    1762              3293   
2          2       3   6353   8808     7684    2405              3516   
3          1       3  13265   1196     4221    6404               507   
4          2       3  22615   5410     7198    3915              1777   
..       ...     ...    ...    ...      ...     ...               ...   
435        1       3  29703  12051    16027   13135               182   
436        1       3  39228   1431      764    4510                93   
437        2       3  14531  15488    30243     437             14841   
438        1       3  10290   1981     2232    1038               168   
439        1       3   2787   1698     2510      65               477   

     Delicassen  
0          1338  
1          1776  
2          7844  
3          1788  
4          5185  
..          ...  
435        2204  
436        2346  
437        1867  
438        2125  
439          52  

[440 rows x 8 columns]

440 records<br>
8 attributes:<br>
FRESH: annual spending (m.u.) on fresh products (Continuous);<br>
MILK: annual spending (m.u.) on milk products (Continuous);<br>
GROCERY: annual spending (m.u.) on grocery products (Continuous);<br>
FROZEN: annual spending (m.u.) on frozen products (Continuous);<br>
DETERGENTS_PAPER: annual spending (m.u.) on detergents and paper products (Continuous); <br>
DELICATESSEN: annual spending (m.u.) on delicatessen products (Continuous); <br>
CHANNEL: customers' channel - Horeca (Hotel/Restaurant/Café) or Retail channel (Nominal);<br>
REGION: customers' region: Lisbon, Oporto or other (Nominal).


Renaming one column with the right name

In [12]:
df.rename(columns={'Delicassen':'Delicatessen'}, inplace = True)

In [13]:
df

Channel  Region  Fresh   Milk  Grocery  Frozen  Detergents_Paper  \
0          2       3  12669   9656     7561     214              2674   
1          2       3   7057   9810     9568    1762              3293   
2          2       3   6353   8808     7684    2405              3516   
3          1       3  13265   1196     4221    6404               507   
4          2       3  22615   5410     7198    3915              1777   
..       ...     ...    ...    ...      ...     ...               ...   
435        1       3  29703  12051    16027   13135               182   
436        1       3  39228   1431      764    4510                93   
437        2       3  14531  15488    30243     437             14841   
438        1       3  10290   1981     2232    1038               168   
439        1       3   2787   1698     2510      65               477   

     Delicatessen  
0            1338  
1            1776  
2            7844  
3            1788  
4            5185  
..            ...  
435          2204  
436          2346  
437          1867  
438          2125  
439            52  

[440 rows x 8 columns]

Nominal attributes as 'Channel' and 'Region' have been encoded.
We know that the frequency of each region are:<br>
Lisbon	77<br>
Oporto	47<br>
Other Region	316<br>

while the frequency of each channel:<br>
Horeca	298 <br>
Retail	142 <br>
Let's see which value the different region and channels have.

In [22]:
print(df['Region'].unique())
print(df['Channel'].unique())

[3 1 2]
[2 1]


In [23]:
R_count_of_1 = df['Region'].value_counts().get(1, df['Region'])
R_count_of_2 = df['Region'].value_counts().get(2, df['Region'])
print(R_count_of_1)
print(R_count_of_2)

77
47


In [24]:
C_count_of_1 = df['Channel'].value_counts().get(1, df['Channel'])
print(C_count_of_1)

298


So:
Lisbon = 1 /
Oporto = 2 /
Other = 3

Horeca = 1 /
Retail = 2